gtk/MakeDialog.cc

```diff
@@ -401,12 +401,27 @@ void MakeDialog::Impl::updatePiecesLabel()
     pieces_lb_->set_text(gstr);
 }
 
-void MakeDialog::Impl::configurePieceSizeScale(uint32_t piece_size)
+void MakeDialog::Impl::configurePieceSizeComboBox(uint32_t piece_size)
 {
-    // the below lower & upper bounds would allow piece size selection between approx 1KiB - 64MiB
-    auto adjustment = Gtk::Adjustment::create(log2(piece_size), 10, 26, 1.0, 1.0);
-    piece_size_scale_->set_adjustment(adjustment);
-    piece_size_scale_->set_visible(true);
+    auto constexpr PieceSizeStart = 14; // 16 KiB
+    auto constexpr PieceSizeEnd = 28; // 256 MiB
+
+    auto active_index = PieceSizeStart;
+    auto options = std::vector<std::pair<Glib::ustring, int>>();
+
+    for (std::size_t i = PieceSizeStart; i <= PieceSizeEnd; i++)
```

Comment:
Should `i` to be an `int` here? It's used for the pair and for `active_index`, both of which hold an `int`.

Reply:
Yes, you are right. Fixed.